crypto/encoding/amino/amino.go

```diff
@@ -60,11 +60,20 @@ func RegisterAmino(cdc *amino.Codec) {
 		secp256k1.PrivKeyAminoName, nil)
 }
 
+// RegisterKeyType registers an external key type to allow decoding it from bytes
+func RegisterKeyType(o interface{}, name string) {
+	cdc.RegisterConcrete(o, name, nil)
+
+	nameTable[reflect.TypeOf(o)] = name
```

Comment:
Should we check if an entry with the same name already exists before possibly overriding it?

Reply:
Yeah, I'll do that now. Do you want me to use the added types to be used in `RegisterAmino`?